tests/tests/shader_primitive_index/mod.rs

```diff
@@ -168,83 +169,31 @@ fn pulling_common(
 
+    let readback_buffer = image::ReadbackBuffers::new(&ctx.device, &color_texture);
+
     let mut encoder = ctx
         .device
         .create_command_encoder(&wgpu::CommandEncoderDescriptor::default());
-
-    let mut rpass = encoder.begin_render_pass(&wgpu::RenderPassDescriptor {
-        color_attachments: &[Some(wgpu::RenderPassColorAttachment {
-            ops: wgpu::Operations {
-                load: wgpu::LoadOp::Clear(wgpu::Color::WHITE),
-                store: true,
-            },
-            resolve_target: None,
-            view: &color_view,
-        })],
-        depth_stencil_attachment: None,
-        label: None,
-    });
-
-    rpass.set_pipeline(&pipeline);
-    rpass.set_index_buffer(index_buffer.slice(..), wgpu::IndexFormat::Uint32);
-    rpass.set_vertex_buffer(0, vertex_buffer.slice(..));
-    function(&mut rpass);
-
-    drop(rpass);
-
-    ctx.queue.submit(Some(encoder.finish()));
-
-    let data = capture_rgba_u8_texture(ctx, color_texture, texture_size);
-
-    assert_eq!(data, expected);
-}
-
-fn capture_rgba_u8_texture(
-    ctx: TestingContext,
-    color_texture: wgpu::Texture,
-    texture_size: wgpu::Extent3d,
-) -> Vec<u8> {
-    let bytes_per_row = align_to(4 * texture_size.width, wgpu::COPY_BYTES_PER_ROW_ALIGNMENT);
-    let buffer_size = bytes_per_row * texture_size.height;
-    let output_buffer = ctx
-        .device
-        .create_buffer_init(&wgpu::util::BufferInitDescriptor {
+    {
+        let mut rpass = encoder.begin_render_pass(&wgpu::RenderPassDescriptor {
+            color_attachments: &[Some(wgpu::RenderPassColorAttachment {
+                ops: wgpu::Operations {
+                    load: wgpu::LoadOp::Clear(wgpu::Color::WHITE),
+                    store: true,
+                },
+                resolve_target: None,
+                view: &color_view,
+            })],
+            depth_stencil_attachment: None,
             label: None,
-            contents: &vec![0u8; buffer_size as usize],
-            usage: wgpu::BufferUsages::COPY_DST | wgpu::BufferUsages::MAP_READ,
         });
 
-    let mut encoder = ctx
-        .device
-        .create_command_encoder(&wgpu::CommandEncoderDescriptor::default());
-
-    encoder.copy_texture_to_buffer(
-        wgpu::ImageCopyTexture {
-            texture: &color_texture,
-            mip_level: 0,
-            origin: wgpu::Origin3d::ZERO,
-            aspect: wgpu::TextureAspect::All,
-        },
-        wgpu::ImageCopyBuffer {
-            buffer: &output_buffer,
-            layout: wgpu::ImageDataLayout {
-                offset: 0,
-                bytes_per_row: Some(bytes_per_row),
-                rows_per_image: None,
-            },
-        },
-        texture_size,
-    );
-
+        rpass.set_pipeline(&pipeline);
+        rpass.set_index_buffer(index_buffer.slice(..), wgpu::IndexFormat::Uint32);
+        rpass.set_vertex_buffer(0, vertex_buffer.slice(..));
+        function(&mut rpass);
```

Comment:
What is this call?

Reply:
So this is the original name of the parameter that was written. It was very descriptive,  but as I was refactoring this function, somehow I didn't notice that.